### PR TITLE
Fix simple/trivial lint issues

### PR DESCRIPTION
**Pull request by joeatwork**
Fix simple lint warnings in advance of release. Fixes logtag length issues, introduces issues with deprecated HTTP tools by upping build target version to 22.

### CODE REVIEW

File: src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java [hunk 1]
```diff
@@ -153,10 +153,10 @@ protected String name() {
          * in the DFS stack already, there must be a circle in the graph. To speed up the
          * search, all the parsed nodes will be removed from the graph.
          */
-        public boolean hasCycle(ArrayMap<View, ArrayList<View>> dependencyGraph) {
-            ArrayList<View> dfsStack = new ArrayList<View>();
+        public boolean hasCycle(TreeMap<View, List<View>> dependencyGraph) {
```
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java]: why TreeMap?

**Comment by yinfeiru** [comment]
Looks great.
One thing I'm not sure of is that if we want to bump the target version to 22. According to Android official site, only 0.7% of the users are on Lollipop 5.1 (the original `target version 21` also looks too critical to me).
http://developer.android.com/about/dashboards/index.html

Can you please give me more insights here?

File: src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java (hunk 1 above)
Comment (joeatwork) [inline src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java]: ArrayMap isn't available in API 16, and I needed another map with a concept of "first element"

File: build.gradle [hunk 2]
```diff
@@ -12,20 +12,21 @@ buildscript {
 }
 
 android {
-    compileSdkVersion 21
-    buildToolsVersion "21.1.2"
+    compileSdkVersion 22
```
Comment (joeatwork) [inline build.gradle]: Discussed, let's hold on changing the target version until after the release.

**Pull request merged**